lib/java/src/org/apache/thrift/transport/TFastFramedTransport.java

```diff
@@ -158,6 +159,10 @@ private void readFrame() throws TTransportException {
 
   @Override
   public void write(byte[] buf, int off, int len) throws TTransportException {
+    if (writeBuffer.getPos() == 0) {
+	// did not put in ctor because it caused interface changes
+        writeBuffer.write(sizeFiller, 0, 4);
```

Comment:
(nit) Indentation is incorrect.

Reply:
This is unfortunate, because it causes an extra conditional check on every `write()` call.

Reply:
Agree, but the alternative was to change all the constructors so they could throw TTransportException.  That's still a viable option but it is a breaking change for some folks who now have to handle the potential (but in all likelihood will never happen) exception.  What do you think it better?  A single size lookup is pretty fast compared to the array copy of writing so I thought this would be acceptable.